src/parser.rs

```diff
@@ -387,56 +371,61 @@ impl Library {
                             ))
                         }
                         "union" => {
-                            #[cfg(feature = "use_unions")]
-                            {
-                                use self::Type::*;
-                                if let Union(mut u) =
-                                    try!(self.read_union_unsafe(parser, ns_id, attrs))
-                                {
-                                    // A nested union->struct->union typically has no c_type
-                                    // so we iterate over fields to find it. These fields are
-                                    // within the nested union->struct if found
-                                    let mut nested = true;
-                                    for f in &mut u.fields {
-                                        if f.c_type.is_none() ||
-                                            c_type == u.c_type.as_ref().unwrap()
-                                        {
-                                            nested = true;
-                                            u.name = format!("{}_u{}", c_type, union_count);
-                                            u.c_type = Some(format!("{}_u{}", c_type, union_count));
-                                        }
+                            if let Type::Union(mut u) = try!(self.read_union(parser, ns_id, attrs)) {
+                                // A nested union->struct->union typically has no c_type
+                                // so we iterate over fields to find it. These fields are
+                                // within the nested union->struct if found
```

Comment:
@Luke-Nukem This part here seems a bit weird btw. Why do we have to iterate over all fields to detect if this union is an anonymous one? Shouldn't we *know* that it's a nested one, because `read_union` is called from `read_union` or `read_record`?